amethyst_core/src/transform/components/isometric_transform.rs

```diff
@@ -0,0 +1,142 @@
+use cgmath::{Angle, Deg, Matrix2, SquareMatrix, Vector2};
+use specs::{Component, DenseVecStorage, FlaggedStorage};
+
+/// Component describing the position of an entity on an isometric plane.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(default)]
+pub struct IsometricTransform {
+    /// Position of the transform on the isometric plane
+    pub translation: Vector2<f32>,
+    /// Weight influencing the likelihood of this entity to be rendered
+    /// on top of other entities. Entities are ordered by `-local_y + priority`.
+    /// The higher the priority, the higher the likelihood of this entity
+    /// to be rendered on top. 
+    pub order_priority: f32,
```

Comment:
I don't know how I feel about this being f32. I personally would've liked it to be either u16 or u32 since you can then use an enum for the ordering which is way more flexible. Maybe this could even be renamed to layer if we go with integers?

Reply:
Well as I'm adding it to an f32 value it removes the casting overhead. And nothing prevents you from making an enum associated with floating point numbers.

Reply:
That's true actually. You can't do `enum_value as f32` but I suppose doing something like `emun_value as isize as f32` upon creation would be the way to go here.

Reply:
cf my response above